instrumentation/graphql/Appraisals

```diff
@@ -4,18 +4,31 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-appraise 'graphql-1.13' do
+# Max compatible version of 1.x
+appraise 'graphql-1.x' do
   gem 'graphql', '~> 1.13'
 end
 
-appraise 'graphql-2.0.17' do
-  gem 'graphql', '2.0.17'
-end
-
+# A bug was introduced in 2.0.18 that was fixed in 2.0.19
 appraise 'graphql-2.0.18' do
   gem 'graphql', '2.0.18'
 end
 
-appraise 'graphql-2.x' do
-  gem 'graphql', '>= 2.0.18', '< 3.0.0'
+# Max compatible version of 2.0.x
+appraise 'graphql-2.0' do
+  gem 'graphql', '~> 2.0.27'
+end
+
+# Max compatible version of 2.1.x
+appraise 'graphql-2.1' do
+  gem 'graphql', '~> 2.1.8'
+end
+
+appraise 'graphql-c_parser-2.2.x' do
+  gem 'graphql', '~> 2.2.1'
+  gem 'graphql-c_parser', '~> 1.0.7'
+end
+
+appraise 'graphql-2.2.x' do
+  gem 'graphql', '~> 2.2.1', '< 3.0.0'
```

Comment:
Did you want to have one run against latest?

Perhaps could match the styles of other appraisals to the same effect.

```suggestion
# Max compatible version of 2.2.x
appraise 'graphql-2.2' do
  gem 'graphql', '~> 2.2.1'
```

Reply:
Major version changes have been difficult for us to test using appraisals, while maintaining minor versions. 

We used to have canary builds that rain on unpinned versions of gems but I couldn't keep up with them so I disabled those builds.

I will however add unpinned versions of the gem to this PR and see if that puts us in a better place for the future canary work.

Reply:
Added here: https://github.com/github/github/discussions/302107